src/Workspaces/Core/Portable/Remote/RemoteHostClient.cs

```diff
@@ -115,5 +115,57 @@ public void Dispose()
                 PinnedScope.Dispose();
             }
         }
+
+        public class NoOpClient : RemoteHostClient
+        {
+            public NoOpClient(Workspace workspace) :
+                base(workspace)
+            {
+            }
+
+            protected override Task<Session> CreateServiceSessionAsync(
+                string serviceName, PinnedRemotableDataScope snapshot, object callbackTarget, CancellationToken cancellationToken)
+            {
+                return Task.FromResult<Session>(new NoOpSession(snapshot, cancellationToken));
+            }
+
+            protected override void OnConnected()
+            {
+                // do nothing
+            }
+
+            protected override void OnDisconnected()
+            {
+                // do nothing
+            }
+
+            private class NoOpSession : Session
+            {
+                public NoOpSession(PinnedRemotableDataScope scope, CancellationToken cancellationToken) :
+                    base(scope, cancellationToken)
+                {
+                }
+
+                public override Task InvokeAsync(string targetName, params object[] arguments)
+                {
+                    return SpecializedTasks.EmptyTask;
+                }
+
+                public override Task<T> InvokeAsync<T>(string targetName, params object[] arguments)
+                {
+                    return SpecializedTasks.Default<T>();
```

Comment:
those are not on in RTW. I confirmed that it is off by default. since now decision is made, all features that want to enable to be run in OOP, need to fullfill this new contact. they need to handle session.InvokeAsync returning default(T) or throwing cancellation exception.